src/main/java/dev/dirs/impl/Windows.java

```diff
@@ -0,0 +1,199 @@
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_CHAR;
+
+public final class Windows {
+
+	private Windows() {}
+
+	static {
+		if (Constants.operatingSystem == 'w') {
+			System.loadLibrary("ole32");
+			System.loadLibrary("shell32");
+		}
+	}
+
+	private static final SymbolLookup SYMBOL_LOOKUP = SymbolLookup.loaderLookup().or(Linker.nativeLinker().defaultLookup());
+	private static final ValueLayout.OfByte C_CHAR = ValueLayout.JAVA_BYTE;
+	private static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT;
+	private static final AddressLayout C_POINTER = ValueLayout.ADDRESS
+			.withTargetLayout(MemoryLayout.sequenceLayout(java.lang.Long.MAX_VALUE, JAVA_BYTE));
+	private static final ValueLayout.OfInt C_LONG = ValueLayout.JAVA_INT;
+
+	public static String getProfileDir() {
+		return getDir("{5E6C858F-0E22-4760-9AFE-EA3317B67173}");
+	}
+
+	public static String getMusicDir() {
+		return getDir("{4BD8D571-6D19-48D3-BE97-422220080E43}");
+	}
+
+	public static String getDesktopDir() {
+		return getDir("{B4BFCC3A-DB2C-424C-B029-7FE99A87C641}");
+	}
+
+	public static String getDocumentsDir() {
+		return getDir("{FDD39AD0-238F-46AF-ADB4-6C85480369C7}");
+	}
+
+	public static String getDownloadsDir() {
+		return getDir("{374DE290-123F-4565-9164-39C4925E467B}");
+	}
+
+	public static String getPicturesDir() {
+		return getDir("{33E28130-4E1E-4676-835A-98395C3BC3BB}");
+	}
+
+	public static String getPublicDir() {
+		return getDir("{DFDF76A2-C82A-4D63-906A-5644AC457385}");
+	}
+
+	public static String getTemplatesDir() {
+		return getDir("{A63293E8-664E-48DB-A079-DF759E0509F7}");
+	}
+
+	public static String getVideosDir() {
+		return getDir("{18989B1D-99B5-455B-841C-AB7C74E4DDFC}");
+	}
+
+	public static String getRoamingAppDataDir() {
+		return getDir("{3EB685DB-65F9-4CF6-A03A-E3EF65729F3D}");
+	}
+
+	public static String getLocalAppDataDir() {
+		return getDir("{F1B32785-6FBA-4FCF-9D55-7B8E7F157091}");
+	}
+
+	public static String applicationPath(String qualifier, String organization, String application) {
+		StringBuilder buf = new StringBuilder(Math.max(stringLength(organization) + stringLength(application), 0));
+		boolean orgPresent = !isNullOrEmpty(organization);
+		boolean appPresent = !isNullOrEmpty(application);
+		if (orgPresent) {
+			buf.append(organization);
+			if (appPresent)
+				buf.append('\\');
+		}
+		if (appPresent)
+			buf.append(application);
+		return buf.toString();
+	}
+
+	private static String getDir(String folderId) {
+		try (var arena = Arena.ofConfined()) {
+			MemorySegment guidSegment = arena.allocate(GUID_LAYOUT);
+			if (CLSIDFromString(createSegmentFromString(folderId, arena), guidSegment) != 0) {
+				throw new AssertionError("failed converting string " + folderId + " to KnownFolderId");
+			}
+			MemorySegment path = arena.allocate(C_POINTER);
+			SHGetKnownFolderPath(guidSegment, 0, MemorySegment.NULL, path);
```

Comment:
Per windows API docs, 
> The calling process is responsible for freeing this resource once it is no longer needed by calling CoTaskMemFree, whether SHGetKnownFolderPath succeeds or not.
Therefore, as currently implemented there would appear to be a minor memory leak.

(see https://learn.microsoft.com/en-us/windows/win32/api/shlobj_core/nf-shlobj_core-shgetknownfolderpath )

Reply:
We can fix this:

Call `CoTaskMemFree(path);` after calling `SHGetKnownFolderPath`.

Add the following:

```java
    private static class CoTaskMemFree {
        public static final FunctionDescriptor DESC = FunctionDescriptor.ofVoid(C_POINTER);

        public static final MethodHandle HANDLE = Linker.nativeLinker().downcallHandle(
                    findOrThrow("CoTaskMemFree"), DESC);
    }
	
	 /**
     * {@snippet lang=c :
     * extern void CoTaskMemFree(LPVOID pv)
     * }
     */
    public static void CoTaskMemFree(MemorySegment pv) {
        var handle = CoTaskMemFree.HANDLE;
        try {
            handle.invokeExact(pv);
        } catch (Throwable throwable) {
           throw new AssertionError("should not reach here", throwable);
        }
    }
```

Reply:
@brcolow Made the changes!
(I'll let the CI run without `System.loadLibrary("combase")` first, will add it too if it complains.)